src/main/java/edu/umich/its/spe/SPEEsbImpl.java

```diff
@@ -0,0 +1,147 @@
+package edu.umich.its.spe;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import edu.umich.ctools.esb.utils.WAPI;
+import edu.umich.ctools.esb.utils.WAPIResultWrapper;
+
+/* 
+ * Implement esb calls to get / put grade information.
+ * The information required is passed in a hashmap.  Some values come from the
+ * properties files.
+ */
+
+// Spring: this makes the class discoverable for autowiring.
+@Component
+
+public class SPEEsbImpl implements SPEEsb {
+
+	static final Logger M_log = LoggerFactory.getLogger(SPEEsbImpl.class);
+
+	// Keys of properties that must be provided for our ESB queries.
+	static final List<String> defaultKeys =  (List<String>) Arrays.asList("tokenServer","apiPrefix","key","secret","scope","x-ibm-client-id");
+
+	/************************** get grades from data warehouse *********************/
+	// Extend the properties list for the properties that get grade requests require.
+	public List<String> setupGetGradePropertyValues() {
+
+		List<String> keys = new ArrayList<>(defaultKeys);
+
+//		// add gradedaftertime as a property for testing and defaulting.
+		keys.add("gradedaftertime");
+		keys.add("COURSEID");
+		keys.add("ASSIGNMENTTITLE");
+
+		return keys;
+	}
+
+	// Go get the SPE grades
+	
+	public WAPIResultWrapper getGradesViaESB(HashMap<String, String> value) throws SPEEsbException {
+		HashMap<String,String> headers = new HashMap<String,String>();
+
+		headers.put("gradedAfterTime",value.get("gradedaftertime"));
+		headers.put("x-ibm-client-id",value.get("x-ibm-client-id"));
+
+		StringBuilder url = new StringBuilder();
+		
+		try {
+			url.append(value.get("apiPrefix"))
+			.append("/Unizin/data/CourseId/")
+			.append(value.get("COURSEID"))
+			.append("/AssignmentTitle/")
+			.append(URLEncoder.encode(value.get("ASSIGNMENTTITLE"),"UTF-8"));
```

Comment:
The current version reads well and there are only two URLs that are constructed.    If there are any more then adding a utility function would be a great idea.